lib/Demangling/Demangler.cpp

```diff
@@ -804,9 +804,14 @@ NodePointer Demangler::demangleType(StringRef MangledName,
 bool Demangler::parseAndPushNodes() {
   const auto textSize = Text.size();
   while (Pos < textSize) {
-    NodePointer Node = demangleOperator();
+    bool foundNULTerminator;
```

Comment:
It's easier if you do the 0-check here instead of in `demangleOperator`:
```
  if (peekChar() == 0) {
    return true;
```
Then you don't need to add the `foundNULTerminator` argument.

Reply:
I was thinking we'd want to keep this next to the other code in `demangleOperator` for consistency. But looking at it again, it doesn't really belong there, since it's just a workaround and not actually decoding an operator. I've changed it to the way you suggest.